controllers/storagecluster/placement.go

```diff
@@ -21,6 +21,11 @@ func getPlacement(sc *ocsv1.StorageCluster, component string) rookv1.Placement {
 		(&in).DeepCopyInto(&placement)
 	}
 
+	// ignore default PodAntiAffinity mon placement when arbiter is enabled
+	if component == "mon" && arbiterEnabled(sc) {
+		placement.PodAntiAffinity = &corev1.PodAntiAffinity{}
```

Comment:
Would this not make it possible for two mons to be deployed on the same node?
The deployment should be like this in stretched cluster mode

1. mon arbiter on a node in arbiter zone
2. 2 mons in zone a such that they are not on the same node
3. 2 mons in zone b such that they are not on the same node

Reply:
the getPlacement function is called one with component name == "arbiter" for mon arbiter
then it is called once with component name == "mon" for the mons placement for zone a and zone b.

Reply:
>> Would this not make it possible for two mons to be deployed on the same node?
That is handled by rook.  Rook adds following podAntiAffinity for mons. So no more than one mon on same node. 
```podAntiAffinity:
      requiredDuringSchedulingIgnoredDuringExecution:
      - labelSelector:
          matchLabels:
            app: rook-ceph-mon
        topologyKey: kubernetes.io/hostname
```

I've added the testing details in the PR description. I was getting the same mon placements in the cluster like you have mentioned above.

Reply:
Thanks for the clarification. Yes, the end result looks good to me.